packages/transaction-controller/src/TransactionController.ts

```diff
@@ -293,6 +292,9 @@ export type TransactionControllerOptions = {
   getGasFeeEstimates?: (
     options: FetchGasFeeEstimateOptions,
   ) => Promise<GasFeeState>;
+  getGlobalProviderAndBlockTracker: () =>
```

Comment:
Heads up: We are trying to phase out passing functions to constructors to get information that essentially comes from other controllers. This has always been a legacy pattern, and as we transition controllers fully to BaseController v2, it's better to make use of the messenger system to access information instead.

The globally selected network is captured in NetworkController as the state property `selectedNetworkClientId`. The combination of provider and block tracker is known as a "network client". Therefore, the pattern that we've been following in other controllers is to allowlist the controller action `NetworkController:getState` and `NetworkController:getNetworkClientById` in the messenger, and then saying:

``` typescript
const { selectedNetworkClientId } = this.messagingSystem.call('NetworkController:getState');
const networkClient = this.messagingSystem.call('NetworkController:getNetworkClientById');
```

Now we can access `networkClient.provider` or `networkClient.blockTracker` as needed. They are guaranteed to point to the same network, since they are created at the same time.

Is it possible to use this pattern for this controller?